### ISSUE TITLE
Some questions about the code

### ISSUE THREAD

**Issue by yrj1409** [argenycw/ILA-DA#2]
1. About interpolation update on X
https://github.com/argenycw/ILA-DA/blob/c7df8ddaad3063d4ae4a722e42f4ee935fa4d4ea/attack.py#L246

**x_adv_if** is obtained by adversarial image without data augmentation. It is inconsistent with the others. According to the paper, I think x_adv_mid and x_mid also should not use data aug?

2. About attack on cifar10

https://github.com/argenycw/ILA-DA/blob/c7df8ddaad3063d4ae4a722e42f4ee935fa4d4ea/attack_cifar.py#L219

the rev_update seems always be **False**. 
Is this technique not effective on CIFAR10 dataset?

**Comment by argenycw** [comment]
Thank you for the questions.

1. About interpolation update on X
`x_adv_if` is computed from `img`, which is obtained from the last image update (i.e. $x''$)

https://github.com/argenycw/ILA-DA/blob/c7df8ddaad3063d4ae4a722e42f4ee935fa4d4ea/attack.py#L236-L237

In Algorithm 1, we can see $\lambda$ is computed **after an effective update of $x''$**. Here $\Delta~y_l''$ no longer takes the input $x''$ (which is from the last iteration), but the newly updated $x''$. Since attack interpolation aims to strengthen $x'$ (`attack_img`), it is performed **AFTER** the update of $img$. The variable name `x_adv_if` means "if we update the attack, what are the intermediate features?" (Not a good name though I admit)

![image](https://user-images.githubusercontent.com/43059631/235906471-2c729408-9662-4d4e-a5fe-d2379807f7ef.png)

In short, `img` does not indicate the augmented version of $x''$, but the updated version (via backprop) of $x''$ in the same iteration.

2. About attack on cifar10
**The whole line should be removed.** That should be an error that we "conveniently" added when we conducted the study internally. The effectiveness of reverse update in CIFAR-10 is similar to that of ImageNet, where usually w/ is better but occasionally wo/ can be better.
Sorry for the confusion. We will commit and push a new version soon.

**Comment by yrj1409** [comment]
In Algorithm 1, when calculating $\Delta y''_l$, the data augmentation shold be applied to $x''$. But the data augmentation proess  seems missed? In other words, the intermediate features should be obtained by $T_z(x'')$, not by $x''$?

https://github.com/argenycw/ILA-DA/blob/c7df8ddaad3063d4ae4a722e42f4ee935fa4d4ea/attack.py#L248

**Comment by argenycw** [comment]
I see what you mean. After updating $x''$, you suggest applying another augmentation to form $T(x'')$ before using it to compute the norm in the feature space. This can promote both the alignments in the image space and the feature space.

Indeed the original attack interpolation only focuses on using the updated $x''$, without further data augmentation. I believe your suggestion would make better coherence between the data augmentation and attack interpolation. 

In terms of implementation, we can first define a new augmentation `t_aug` by extracting the inline augmentation in 219-220.
https://github.com/argenycw/ILA-DA/blob/c7df8ddaad3063d4ae4a722e42f4ee935fa4d4ea/attack.py#L219-L220

After that we change 
https://github.com/argenycw/ILA-DA/blob/c7df8ddaad3063d4ae4a722e42f4ee935fa4d4ea/attack.py#L246
into
```py
x_adv_if = ila_forw_by_models(model, model_type, t_aug(img), ila_layer)  
```

Thank you for pointing out the potential improvements. In short, the result reported in the paper is the current version but I do agree your suggestion makes more sense and will potentially improve the attack. To let the readers reproduce the reported results, I might not change the master branch in a short period of time. But it will definitely be adapted in future updates.

**Comment by yrj1409** [comment]
okay, thanks for your reply!
